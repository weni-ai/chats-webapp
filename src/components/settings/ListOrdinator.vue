<template>
  <div class="order">
    <p
      v-if="label"
      class="label"
    >
      {{ label }}:
    </p>

    <section class="ordinators">
      <UnnnicRadio
        v-for="(ordinator, index) in ordinators"
        :key="index"
        size="md"
        :value="ordinator"
        :modelValue="modelValue"
        @update:model-value="$emit('update:model-value', $event)"
      >
        {{ $t(`order_by.option.${ordinator}`) }}
      </UnnnicRadio>
    </section>
  </div>
</template>

<script>
export default {
  props: {
    label: { type: String, default: '' },
    ordinators: {
      type: Array,
      default: () => ['alphabetical', 'newer', 'older'],
    },
    modelValue: { type: String, required: true },
  },
  emits: ['update:model-value'],
};
</script>

<style lang="scss" scoped>
.order {
  display: flex;
  align-items: center;
  flex-wrap: wrap;
  font-family: $unnnic-font-family-secondary;
  font-weight: $unnnic-font-weight-regular;
  color: $unnnic-color-neutral-darkest;
  font-size: $unnnic-font-size-body-gt;
  line-height: $unnnic-font-size-body-gt + $unnnic-line-height-md;
  column-gap: $unnnic-spacing-sm;
  row-gap: $unnnic-spacing-stack-nano;

  .ordinators {
    display: flex;
    align-items: center;
    column-gap: $unnnic-spacing-inline-xs;
  }
}
</style>
