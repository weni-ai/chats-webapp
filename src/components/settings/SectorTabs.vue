<template>
  <UnnnicTab
    v-model:activeTab="tab"
    :initialTab="tab || 'sector'"
    :tabs="tabs"
    class="sector-tabs"
    @change="tab = $event"
  >
    <template #tab-head-sector>
      <div class="form-tab">
        <span class="name">{{ $t('sector.title') }}</span>
      </div>
    </template>

    <template #tab-head-queues>
      <div class="form-tab">
        <span class="name">{{ $t('queues.title') }}</span>
      </div>
    </template>

    <template #tab-head-messages>
      <div class="form-tab">
        <span class="name">{{ $t('settings.messages.title') }}</span>
      </div>
    </template>

    <template #tab-head-tags>
      <div class="form-tab">
        <span class="name">{{ $t('tags.title') }}</span>
      </div>
    </template>

    <template #tab-panel-sector>
      <slot name="sector" />
    </template>

    <template #tab-panel-queues>
      <slot name="queues" />
    </template>

    <template #tab-panel-messages>
      <slot name="messages" />
    </template>

    <template #tab-panel-tags>
      <slot name="tags" />
    </template>
  </UnnnicTab>
</template>

<script>
export default {
  name: 'SectorTabs',

  props: {
    modelValue: {
      type: String,
      default: '',
    },
  },

  data: () => ({
    tabs: ['sector', 'queues', 'messages', 'tags'],
  }),

  computed: {
    tab: {
      get() {
        return this.modelValue;
      },
      set(tab) {
        if (typeof tab === 'string') this.$emit('update:modelValue', tab);
      },
    },
  },

  mounted() {
    const { tab } = this.$route.query;

    if (this.tabs.includes(tab)) this.tab = tab;
  },
};
</script>

<style lang="scss" scoped>
.sector-tabs {
  .form-tab {
    display: flex;
    align-items: center;
    gap: $unnnic-spacing-stack-xs;

    .name {
      font-size: $unnnic-font-size-body-lg;
    }
  }
}
</style>
