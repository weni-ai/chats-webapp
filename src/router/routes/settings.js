const routes = [
  {
    path: '/settings',
    component: () => import('@/layouts/SettingsLayout/index.vue'),
    children: [
      {
        path: '/settings',
        name: 'sectors',
        component: () => import('@/views/Settings/index.vue'),
      },

      {
        path: 'sectors/new',
        name: 'sectors.new',
        component: () => import('@/views/Settings/Sectors/New.vue'),
      },
      {
        path: 'sectors/:uuid',
        name: 'sectors.edit',
        component: () => import('@/views/Settings/Sectors/Edit.vue'),
        props: (route) => ({
          uuid: route.params.uuid,
          tab: route.query.tab,
        }),
      },
    ],
  },
];

export default routes;
