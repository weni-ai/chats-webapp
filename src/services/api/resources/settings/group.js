import http from '@/services/api/http';
import { getProject } from '@/utils/config';

export default {
  async list(params = {}) {
    const endpoint = '/group_sector/';

    const response = await http.get(endpoint, {
      params: { ...params, project: getProject() },
    });

    return response.data;
  },

  async create({ body }) {
    const endpoint = `/group_sector/`;
    const response = await http.post(endpoint, body);

    return response.data;
  },

  async update({ groupUuid, body }) {
    const endpoint = `/group_sector/${groupUuid}/`;
    const response = await http.patch(endpoint, body);

    return response.data;
  },

  async addSector({ groupUuid, sectorUuid }) {
    const endpoint = `/group_sector/${groupUuid}/add_sector/`;
    const response = await http.post(endpoint, { sector: sectorUuid });

    return response.data;
  },

  async removeSector({ groupUuid, sectorUuid }) {
    const endpoint = `/group_sector/${groupUuid}/remove_sector/`;
    const response = await http.post(endpoint, { sector: sectorUuid });

    return response.data;
  },

  async listAuthorization({ groupSectorUuid, role }) {
    const endpoint = `/authorization/group_sector/`;
    const params = { group_sector: groupSectorUuid, role };

    const response = await http.get(endpoint, { params });

    return response.data;
  },

  async addAuthorization({ groupSectorUuid, permissionUuid, role }) {
    const endpoint = `/authorization/group_sector/`;
    const body = {
      role,
      permission: permissionUuid,
      group_sector: groupSectorUuid,
    };

    const response = await http.post(endpoint, body);

    return response.data;
  },
};
