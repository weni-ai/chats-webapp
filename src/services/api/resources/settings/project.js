import http from '@/services/api/http';
import { getProject } from '@/utils/config';

export const Permissions = Object.freeze({
  Admin: 1,
  Agent: 2,
  Manager: 3,
});

export default {
  async admins() {
    const response = await http.get(`/permission/project/`, {
      params: {
        project: getProject(),
        // role: Permissions.Admin,
        limit: 9999,
      },
    });
    return response.data;
  },

  async managers(offset, limit = 20) {
    const response = await http.get(`/permission/project/`, {
      params: {
        project: getProject(),
        offset,
        // role: Permissions.Manager,
        limit,
      },
    });
    return response.data;
  },

  async agents() {
    const response = await http.get(`/permission/project/`, {
      params: {
        project: getProject(),
        // role: Permissions.Agent,
        limit: 9999,
      },
    });
    return response.data;
  },

  async allUsers() {
    const response = await http.get(`/project/${getProject()}/list_users`, {
      params: {
        limit: 9999,
      },
    });
    return response.data;
  },

  async getInfo() {
    const projectUuid = getProject();
    const response = await http.get(`/project/${projectUuid}/`);
    return response;
  },

  async update(data) {
    const projectUuid = getProject();
    const body = {
      config: JSON.stringify(data),
    };

    const response = await http
      .patch(`/project/${projectUuid}/`, body)
      .then((response) => response.data)
      .catch((error) => error.response);
    return response;
  },
};
