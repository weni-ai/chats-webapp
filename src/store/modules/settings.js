import { defineStore } from 'pinia';
import Sector from '@/services/api/resources/settings/sector';
import cloneDeep from 'lodash.clonedeep';
import { removeDuplicatedItems } from '@/utils/array';

export const useSettings = defineStore('settings', {
  state: () => ({
    sectors: [],
    isLoadingSectors: false,
    nextSectors: '',
    previousSectors: '',
    currentSector: null,
  }),

  actions: {
    async getSectors() {
      const isInLastPage = !this.nextSectors && this.previousSectors;
      if (this.isLoadingSectors || isInLastPage) {
        return;
      }

      try {
        this.isLoadingSectors = true;
        const { results, next, previous } = await Sector.list({
          nextReq: this.nextSectors,
        });
        this.sectors = removeDuplicatedItems([...this.sectors, ...results]);
        this.nextSectors = next;
        this.previousSectors = previous;
      } catch (error) {
        console.error(error);
      } finally {
        this.isLoadingSectors = false;
      }
    },

    addSector(sector) {
      const lastId = this.sectors.at(-1).id;
      let queueId = 100;
      this.sectors.push({
        ...sector,
        id: lastId + 1,
        queues: sector.queues.map((q) => ({ ...q, id: queueId++ })),
      });
    },

    async getCurrentSector(uuid) {
      this.currentSector = await Sector.find(uuid);
    },

    saveSector(sector) {
      this.addSector({ ...sector, contacts: { count: 0 } });
    },

    updateSector(sector) {
      if (!sector.id) this.saveSector(sector);
      const index = this.sectors.findIndex((s) => s.id === sector.id);
      this.sectors.splice(index, 1, sector);
    },

    async deleteSector(sectorUuid) {
      await Sector.deleteSector(sectorUuid);
      this.sectors = this.sectors.filter(
        (sector) => sector.uuid !== sectorUuid,
      );
    },
  },

  getters: {
    getSectorById({ sectors }) {
      return (id) => {
        const sector = sectors.find((sector) => sector.id === id);
        return sector ? cloneDeep(sector) : null;
      };
    },
  },
});
