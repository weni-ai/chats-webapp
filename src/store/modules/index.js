import auth from './auth';
import chats from './chats';
import settings from './settings';
import rooms from './rooms';

export default {
  auth,
  chats,
  settings,
  rooms,
};
