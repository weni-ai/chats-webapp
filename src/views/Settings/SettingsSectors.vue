<template>
  <SettingsSectionHeader
    data-testid="settings-sectors-header"
    :title="$t('config_chats.section_sectors_title', { project: projectName })"
    :subtitle="$t('config_chats.section_sectors_subtitle')"
  />
  <section class="settings-sectors__filters">
    <UnnnicInput
      v-model="sectorNameFilter"
      iconLeft="search-1"
      size="md"
      :placeholder="$t('search')"
    />
    <ListOrdinator
      v-model="sectorOrder"
      :label="$t('order_by.label')"
    />
  </section>

  <section class="settings-view__sectors">
    <UnnnicCard
      data-testid="settings-sectors-blank-card"
      type="blank"
      :text="$t('config_chats.new_sector')"
      icon="add"
      class="sectors__new-sector-card"
      @click="openNewSectorModal()"
    />

    <UnnnicSimpleCard
      v-for="sector in sectorsOrdered"
      :key="sector.id"
      :title="sector.name"
      clickable
      class="sectors__card"
      data-testid="settings-sectors-sector-card"
      @click="navigate('sectors.edit', { uuid: sector.uuid })"
    >
      <template #headerSlot>
        <UnnnicDropdown position="top-left">
          <template #trigger>
            <UnnnicToolTip
              enabled
              :text="$t('quick_messages.delete_or_edit')"
              side="left"
            >
              <UnnnicButton
                iconCenter="more_vert"
                type="tertiary"
                data-testid="open-dropdown-menu-button"
              />
            </UnnnicToolTip>
          </template>
          <UnnnicDropdownItem
            data-testid="dropdown-edit"
            @click="navigate('sectors.edit', { uuid: sector.uuid })"
          >
            <section class="dropdown-item-content">
              <UnnnicIconSvg
                class="icon"
                icon="edit_square"
                size="sm"
              />
              <p>{{ $t('edit') }}</p>
            </section>
          </UnnnicDropdownItem>
          <UnnnicDropdownItem
            data-testid="dropdown-delete"
            @click.stop="handlerOpenDeleteSectorModal(sector)"
          >
            <section
              class="dropdown-item-content dropdown-item-content__delete"
            >
              <UnnnicIconSvg
                class="icon"
                icon="delete"
                size="sm"
                scheme="danger"
              />
              <p>{{ $t('exclude') }}</p>
            </section>
          </UnnnicDropdownItem>
        </UnnnicDropdown>
      </template>
    </UnnnicSimpleCard>
  </section>

  <section
    v-if="isLoadingSectors"
    data-testid="settings-sectors-loading-section"
    class="settings-view__loading-sectors"
  >
    <img
      src="@/assets/LogoWeniAnimada4.svg"
      width="40"
    />
  </section>
  <NewSectorDrawer
    v-if="showNewSectorModal"
    v-model="showNewSectorModal"
    data-testid="new-sector-drawer"
    @close="closeNewSectorModal()"
  />
  <UnnnicModalNext
    v-if="showDeleteSectorModal"
    type="alert"
    icon="alert-circle-1"
    scheme="feedback-red"
    :title="$t('delete_sector') + ` ${toDeleteSector.name}`"
    :description="$t('cant_revert')"
    :validate="`${toDeleteSector.name}`"
    :validatePlaceholder="`${toDeleteSector.name}`"
    :validateLabel="
      $t('confirm_typing') + ` &quot;${toDeleteSector.name}&quot;`
    "
    :actionPrimaryLabel="$t('confirm')"
    :actionSecondaryLabel="$t('cancel')"
    data-testid="modal-delete-sector"
    @click-action-primary="deleteSector(toDeleteSector.uuid)"
    @click-action-secondary="handlerCloseDeleteSectorModal()"
  />
</template>

<script>
import unnnic from '@weni/unnnic-system';
import { mapActions, mapState } from 'pinia';

import { useConfig } from '@/store/modules/config';
import { useSettings } from '@/store/modules/settings';

import SettingsSectionHeader from './SettingsSectionHeader.vue';

import NewSectorDrawer from './Sectors/New/NewSectorDrawer.vue';
import ListOrdinator from '@/components/settings/ListOrdinator.vue';

export default {
  name: 'SettingsSectors',

  components: {
    SettingsSectionHeader,
    NewSectorDrawer,
    ListOrdinator,
  },

  data() {
    return {
      showNewSectorModal: false,
      showDeleteSectorModal: false,
      sectorNameFilter: '',
      sectorOrder: 'alphabetical',
      toDeleteSector: {},
    };
  },

  computed: {
    ...mapState(useConfig, {
      projectName: (store) => store.project.name,
    }),
    ...mapState(useSettings, ['sectors', 'isLoadingSectors']),

    sectorsOrdered() {
      let sectorsOrdered = this.sectors.slice().sort((a, b) => {
        let first = null;
        let second = null;

        if (this.sectorOrder === 'alphabetical') {
          first = a.name.toLowerCase();
          second = b.name.toLowerCase();
        } else if (this.sectorOrder === 'newer') {
          first = new Date(b.created_on).getTime();
          second = new Date(a.created_on).getTime();
        } else if (this.sectorOrder === 'older') {
          first = new Date(a.created_on).getTime();
          second = new Date(b.created_on).getTime();
        }

        return first === second ? 0 : first > second ? 1 : -1;
      });

      return this.sectorNameFilter.trim()
        ? sectorsOrdered.filter(({ name }) =>
            name
              .toLowerCase()
              .includes(this.sectorNameFilter.trim().toLowerCase()),
          )
        : sectorsOrdered;
    },
  },

  methods: {
    ...mapActions(useSettings, {
      actionDeleteSector: 'deleteSector',
    }),
    handlerOpenDeleteSectorModal(sector) {
      this.toDeleteSector = sector;
      this.handleConnectOverlay(true);
      this.showDeleteSectorModal = true;
    },

    handlerCloseDeleteSectorModal() {
      this.toDeleteSector = {};
      this.handleConnectOverlay(false);
      this.showDeleteSectorModal = false;
    },

    async deleteSector(sectorUuid) {
      try {
        await this.actionDeleteSector(sectorUuid);

        this.$router.push({ name: 'sectors' });
        unnnic.unnnicCallAlert({
          props: {
            text: this.$t('sector_deleted_success'),
            type: 'success',
          },
          seconds: 5,
        });
      } catch (error) {
        console.log(error);
        unnnic.unnnicCallAlert({
          props: {
            text: this.$t('sector_delete_error'),
            type: 'error',
          },
          seconds: 5,
        });
      } finally {
        this.showDeleteSectorModal = false;
        this.handleConnectOverlay(false);
      }
    },
    handleConnectOverlay(active) {
      window.parent.postMessage({ event: 'changeOverlay', data: active }, '*');
    },
    openNewSectorModal() {
      setTimeout(() => {
        this.handleConnectOverlay(true);
      }, 1);
      this.showNewSectorModal = true;
    },
    closeNewSectorModal() {
      setTimeout(() => {
        this.handleConnectOverlay(false);
      }, 1);
      this.showNewSectorModal = false;
    },
    navigate(name, params) {
      this.$router.push({
        name,
        params,
      });
    },
  },
};
</script>

<style lang="scss" scoped>
.settings-sectors {
  &__filters {
    display: flex;
    flex-wrap: wrap;
    align-items: center;
    gap: $unnnic-spacing-stack-sm $unnnic-spacing-inline-md;

    .unnnic-form {
      flex: 1;
    }
  }
}
.settings-view {
  &__sectors {
    display: grid;
    grid-template-columns: 1fr 1fr 1fr;
    gap: $unnnic-spacing-sm;

    .sectors__card {
      min-height: 120px;

      :deep(.unnnic-simple-card-header-container__title) {
        color: $unnnic-color-neutral-darkest;
      }
    }
  }

  .sectors__new-sector-card:hover {
    box-shadow: $unnnic-shadow-level-far;
  }
  .sectors__new-sector-card:active {
    border: 1px solid $unnnic-color-neutral-cleanest;
  }
  .sectors__new-sector-card {
    min-height: 120px;
    :deep(.unnnic-card-blank__content) {
      flex-direction: row;
    }
    :deep(.unnnic-card-blank__content__icon) {
      font-size: $unnnic-font-size-title-md;
    }
  }

  .dropdown-item-content {
    display: flex;
    align-items: center;
    gap: $unnnic-spacing-xs;

    white-space: nowrap;

    &__delete {
      color: red;
    }
  }

  &__loading-sectors {
    display: flex;
    align-items: center;
    justify-content: center;
    padding: $unnnic-spacing-sm;
  }
}
</style>
