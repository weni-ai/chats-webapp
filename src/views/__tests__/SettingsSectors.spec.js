import { mount } from '@vue/test-utils';
import { describe, it, expect, vi } from 'vitest';

import SettingsSectors from '@/views/Settings/SettingsSectors.vue';

import { createTestingPinia } from '@pinia/testing';
import { useSettings } from '@/store/modules/settings';

const store = createTestingPinia({
  initialState: {
    config: {
      project: { name: 'Project 1' },
    },
    settings: {
      sectors: [
        {
          id: 1,
          uuid: '123',
          name: 'Sector 1',
          agents: 10,
          contacts: 50,
        },
      ],
      isLoadingSectors: false,
    },
  },
});

const createWrapper = (props) => {
  return mount(SettingsSectors, {
    props,
    global: {
      plugins: [store],
      stubs: {
        UnnnicCard: true,
        UnnnicCardProject: true,
        NewSectorDrawer: true,
      },
      mocks: {
        $router: {
          push: vi.fn(),
        },
      },
    },
  });
};

describe('SettingsHeader.vue', () => {
  let wrapper;
  let settingsStore;

  beforeEach(() => {
    settingsStore = useSettings();

    wrapper = createWrapper();
  });

  it('should render the SettingsSectionHeader with correct props', () => {
    const header = wrapper.findComponent(
      '[data-testid=settings-sectors-header]',
    );

    expect(header.exists()).toBe(true);
    expect(header.props('title')).toBe(
      wrapper.vm.$t('config_chats.section_sectors_title', {
        project: 'Project 1',
      }),
    );
    expect(header.props('subtitle')).toBe(
      wrapper.vm.$t('config_chats.section_sectors_subtitle'),
    );
  });

  it('should render the card for creating a new sector', async () => {
    const newSectorCard = wrapper.findComponent(
      '[data-testid=settings-sectors-blank-card]',
    );

    expect(newSectorCard.exists()).toBe(true);
    expect(newSectorCard.props('text')).toBe(
      wrapper.vm.$t('config_chats.new_sector'),
    );
    expect(newSectorCard.props('icon')).toBe('add');

    newSectorCard.trigger('click');

    await wrapper.vm.$nextTick();

    const newSectorDrawer = wrapper.find('[data-testid="new-sector-drawer"]');
    expect(newSectorDrawer.exists()).toBe(true);
  });

  it('should render the correct number of card components', () => {
    const sectorCards = wrapper.findAllComponents(
      '[data-testid=settings-sectors-sector-card]',
    );

    expect(sectorCards.length).toBe(1);

    const sectorCard = sectorCards[0];
    expect(sectorCard.props('name')).toBe('Sector 1');

    sectorCard.trigger('action');
    expect(wrapper.vm.$router.push).toHaveBeenCalledWith({
      name: 'sectors.edit',
      params: { uuid: '123' },
    });
  });

  it('should display the loading section when sectors are loading', async () => {
    settingsStore.isLoadingSectors = true;

    await wrapper.vm.$nextTick();

    const loadingSection = wrapper.find(
      '[data-testid=settings-sectors-loading-section]',
    );

    expect(loadingSection.exists()).toBe(true);
    expect(loadingSection.find('img').attributes('src')).toContain(
      'LogoWeniAnimada4.svg',
    );
  });

  it('should match the snapshot', () => {
    expect(wrapper.element).toMatchSnapshot();
  });
});
