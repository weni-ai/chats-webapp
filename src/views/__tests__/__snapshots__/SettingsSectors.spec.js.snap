// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`SettingsHeader.vue > should match the snapshot 1`] = `
<div
  data-v-app=""
>
  
  <header
    class="settings-view__section-header"
    data-testid="settings-sectors-header"
    data-v-1088b44f=""
    data-v-f23b63f9=""
  >
    <p
      class="section-header__title"
      data-testid="section-header-title"
      data-v-f23b63f9=""
    >
      Sectors in Project 1
    </p>
    <p
      class="section-header__subtitle"
      data-testid="section-header-subtitle"
      data-v-f23b63f9=""
    >
      Add, view and manage sectors, queues, managers and agents within your organization.
    </p>
  </header>
  <section
    class="settings-sectors__filters"
    data-v-1088b44f=""
  >
    <div
      class="unnnic-form md"
      data-v-1088b44f=""
      data-v-24f2d284=""
    >
      <!---->
      <div
        class="text-input size--md text-input--icon-right-size-sm unnnic-form-input"
        data-v-24f2d284=""
        data-v-c1401a33=""
        mask=""
      >
        <input
          class="unnnic-form-input input-itself input size-md normal input--has-icon-left unnnic-form-input input-itself"
          data-v-0caa2ccb=""
          data-v-c1401a33=""
          hascloudycolor="false"
          iconleft="search-1"
          iconrightsize="sm"
          placeholder="Search"
          type="text"
          value=""
        />
        <svg
          class="unnnic-icon unnnic-icon__size--sm unnnic-icon-scheme--neutral-cloudy icon-left"
          data-testid="old-map-icons"
          data-v-9846e6db=""
          data-v-c1401a33=""
          fill="none"
          height="40"
          viewBox="0 0 40 40"
          width="40"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            class="primary"
            d="M34.0615 35.0005C33.8116 35.0005 33.5754 34.903 33.3992 34.7256L25.2575 26.5826C24.3065 27.3961 23.2355 28.0597 22.067 28.5559C20.5312 29.2082 18.9178 29.5394 17.272 29.5394C15.7011 29.5394 14.1578 29.2382 12.6819 28.6421C9.64389 27.4161 7.26574 25.0817 5.98481 22.0662C4.70388 19.052 4.67263 15.7178 5.89858 12.6811C7.12452 9.64433 9.45894 7.26492 12.4732 5.98399C14.009 5.33041 15.6211 5.00049 17.2682 5.00049C18.8391 5.00049 20.3825 5.30291 21.8583 5.89776C24.8963 7.12371 27.2745 9.45812 28.5554 12.4736C29.8363 15.4891 29.8676 18.822 28.6416 21.8588C28.1393 23.1035 27.4469 24.2444 26.5809 25.2554L34.7251 33.4009C35.09 33.7658 35.09 34.3606 34.7251 34.7268C34.5477 34.903 34.3115 35.0005 34.0615 35.0005ZM17.2682 6.87502C15.8748 6.87502 14.5077 7.15495 13.2068 7.70856C10.6524 8.79454 8.67413 10.809 7.63564 13.3821C6.59715 15.9552 6.6234 18.7783 7.70813 21.3327C8.7941 23.887 10.8086 25.8653 13.3817 26.9038C14.6326 27.4086 15.9411 27.6636 17.2707 27.6636C18.6641 27.6636 20.0313 27.3836 21.3322 26.83C22.5494 26.3127 23.6404 25.5828 24.5739 24.6606C24.5864 24.6431 24.6002 24.6281 24.6139 24.6143C24.6339 24.5943 24.6526 24.5781 24.6701 24.5656C25.6236 23.6021 26.381 22.4499 26.9021 21.1565C27.9406 18.5833 27.9143 15.7603 26.8283 13.2059C25.7436 10.6516 23.7279 8.67332 21.1548 7.63483C19.9063 7.13121 18.5979 6.87502 17.2682 6.87502Z"
            fill="#3B414D"
          />
        </svg>
        <!---->
      </div>
      <!---->
    </div>
    <div
      class="order"
      data-v-1088b44f=""
      data-v-348024a5=""
    >
      <p
        class="label"
        data-v-348024a5=""
      >
        Order by: 
      </p>
      <section
        class="ordinators"
        data-v-348024a5=""
      >
        
        <div
          class="unnnic-radio-container unnnic-radio-container--md"
          data-v-348024a5=""
          data-v-abd1b195=""
        >
          <svg
            class="unnnic-icon unnnic-icon__size--md unnnic-icon-scheme--brand-weni unnnic-radio"
            data-testid="old-map-icons"
            data-v-9846e6db=""
            data-v-abd1b195=""
            fill="none"
            height="40"
            viewBox="0 0 40 40"
            width="40"
            xmlns="http://www.w3.org/2000/svg"
          >
            <circle
              class="primary"
              cx="20"
              cy="20.0005"
              fill="#3B414D"
              r="15"
            />
            <circle
              cx="20"
              cy="20.0005"
              fill="white"
              r="6.66667"
            />
          </svg>
          <span
            class="label"
            data-v-abd1b195=""
          >
            
            Alphabetical
            
          </span>
        </div>
        <div
          class="unnnic-radio-container unnnic-radio-container--md"
          data-v-348024a5=""
          data-v-abd1b195=""
        >
          <svg
            class="unnnic-icon unnnic-icon__size--md unnnic-icon-scheme--neutral-cleanest unnnic-radio"
            data-testid="old-map-icons"
            data-v-9846e6db=""
            data-v-abd1b195=""
            fill="none"
            height="40"
            viewBox="0 0 40 40"
            width="40"
            xmlns="http://www.w3.org/2000/svg"
          >
            <circle
              class="primary-stroke"
              cx="20"
              cy="20.0005"
              r="14.5"
              stroke="#3B414D"
            />
          </svg>
          <span
            class="label"
            data-v-abd1b195=""
          >
            
            Newer
            
          </span>
        </div>
        <div
          class="unnnic-radio-container unnnic-radio-container--md"
          data-v-348024a5=""
          data-v-abd1b195=""
        >
          <svg
            class="unnnic-icon unnnic-icon__size--md unnnic-icon-scheme--neutral-cleanest unnnic-radio"
            data-testid="old-map-icons"
            data-v-9846e6db=""
            data-v-abd1b195=""
            fill="none"
            height="40"
            viewBox="0 0 40 40"
            width="40"
            xmlns="http://www.w3.org/2000/svg"
          >
            <circle
              class="primary-stroke"
              cx="20"
              cy="20.0005"
              r="14.5"
              stroke="#3B414D"
            />
          </svg>
          <span
            class="label"
            data-v-abd1b195=""
          >
            
            Older
            
          </span>
        </div>
        
      </section>
    </div>
  </section>
  <section
    class="settings-view__sectors"
    data-v-1088b44f=""
  >
    <unnnic-card-stub
      buttonaction="[Function]"
      class="sectors__new-sector-card"
      clickable="false"
      data-testid="settings-sectors-blank-card"
      data-v-1088b44f=""
      enabled="true"
      hasinformationicon="true"
      icon="add"
      infoposition="right"
      invertedpercentage="false"
      percent="0"
      rating="0"
      scheme="aux-blue"
      text="New sector"
      type="blank"
    />
    
    <section
      class="unnnic-simple-card unnnic-simple-card--clickable sectors__card"
      data-testid="settings-sectors-sector-card"
      data-v-1088b44f=""
      data-v-80dbbab7=""
    >
      <section
        class="unnnic-simple-card__header"
        data-v-80dbbab7=""
      >
        <section
          class="unnnic-simple-card-header-container"
          data-v-80dbbab7=""
        >
          <h1
            class="unnnic-simple-card-header-container__title"
            data-v-80dbbab7=""
          >
            Sector 1
          </h1>
          <!---->
        </section>
        <section
          class="unnnic-simple-card__header-slot"
          data-v-80dbbab7=""
        >
          
          <div
            class="unnnic-dropdown"
            data-v-1088b44f=""
            data-v-36eb394f=""
          >
            <div
              class="unnnic-dropdown__trigger"
              data-v-36eb394f=""
            >
              
              <div
                class="unnnic-tooltip"
                data-v-1088b44f=""
                data-v-8c61d974=""
              >
                
                <button
                  class="unnnic-button unnnic-button--size-large unnnic-button--tertiary unnnic-button--icon-on-center"
                  data-testid="open-dropdown-menu-button"
                  data-v-1088b44f=""
                  data-v-d127031e=""
                >
                  <!---->
                  <!---->
                  <span
                    class="material-symbols-rounded unnnic-icon-scheme--neutral-dark unnnic-icon-size--md"
                    data-testid="icon-center"
                    data-v-9846e6db=""
                    data-v-d127031e=""
                    translate="no"
                  >
                    more_vert
                  </span>
                  <span
                    class="unnnic-button__label"
                    data-testid="button-label"
                    data-v-d127031e=""
                  >
                    
                    
                     
                  </span>
                  <!---->
                </button>
                
                <span
                  class="unnnic-tooltip-label unnnic-tooltip-label-left"
                  data-testid="tooltip-label"
                  data-v-8c61d974=""
                  style="left: -8px; top: 0px;"
                >
                  
                  
                  Delete or edit 
                  <br
                    data-v-8c61d974=""
                  />
                  
                  
                  <!---->
                </span>
              </div>
              
              <!---->
            </div>
          </div>
          
        </section>
      </section>
      <!---->
      <!---->
    </section>
    
  </section>
  <section
    class="settings-view__loading-sectors"
    data-testid="settings-sectors-loading-section"
    data-v-1088b44f=""
  >
    <img
      data-v-1088b44f=""
      src="/src/assets/LogoWeniAnimada4.svg"
      width="40"
    />
  </section>
  <!--v-if-->
  <!--v-if-->
  
</div>
`;
